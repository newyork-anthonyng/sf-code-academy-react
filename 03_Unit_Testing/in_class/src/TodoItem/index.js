import React from "react";
import PropTypes from "prop-types";

const TodoItem = function(props) {
  return (
    <li
      onClick={() => props.onClick(props.index)}
      style={{
        textDecoration: props.completed ? "line-through" : "",
        color: props.completed ? "green" : "black",
        cursor: "pointer"
      }}
    >
      {props.text}
    </li>
  );
};
const a = {
  text: PropTypes.string,
  completed: PropTypes.bool,
  onClick: PropTypes.func,
  index: PropTypes.number
};
TodoItem.propTypes = a;

export default TodoItem;
