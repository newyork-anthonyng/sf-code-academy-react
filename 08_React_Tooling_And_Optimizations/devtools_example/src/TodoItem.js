import React from "react";
import PropTypes from "prop-types";

const TodoItem = ({ text, completed, onClick }) => (
  <li
    style={{ textDecoration: completed && "line-through" }}
    onClick={onClick}
  >
    {text}
  </li>
);

TodoItem.defaultProps = {
  onClick: () => {}
};

TodoItem.propTypes = {
  text: PropTypes.string.isRequired,
  completed: PropTypes.bool,
  onClick: PropTypes.func.isRequired
};

export default TodoItem;
