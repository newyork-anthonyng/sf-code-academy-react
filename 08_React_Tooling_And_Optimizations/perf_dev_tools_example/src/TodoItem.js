import React from "react";
import PropTypes from "prop-types";

class TodoItem extends React.Component {
  shouldComponentUpdate() {
    return false;
  }

  render() {
    const { text, completed, onClick } = this.props;
    return (
      <li
        style={{
          textDecoration: completed && "line-through"
        }}
        onClick={onClick}
      >
        {text}
      </li>
    );
  }
}

TodoItem.defaultProps = {
  onClick: () => {}
};

TodoItem.propTypes = {
  text: PropTypes.string.isRequired,
  completed: PropTypes.bool,
  onClick: PropTypes.func.isRequired
};

export default TodoItem;
