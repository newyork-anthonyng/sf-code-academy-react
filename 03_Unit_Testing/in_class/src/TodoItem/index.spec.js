import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";
import TodoItem from "./";

Enzyme.configure({ adapter: new Adapter() });

// Add new tests here
it("should return true", function() {
  expect(true).toEqual(true);
});

const array = [
  { name: "The Dark Knight", rating: 7, ageRating: 12 },
  { name: "Up", rating: 10, ageRating: 14 },
];

it("should have correct movie", function() {
  expect(array).toMatchSnapshot();
});

const defaultProps = {
  completed: false,
  onClick: function() {},
  text: "Walk dog",
  index: 42
};

it("should render correctly when completed", function() {
  const wrapper = shallow(
    <TodoItem {...defaultProps} completed={true} />
  );

  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("should render correctly when not completed", function() {
  const wrapper = shallow(
    <TodoItem {...defaultProps} />
  );

  expect(toJSON(wrapper)).toMatchSnapshot();
  expect({ foo: "bar" }).toMatchSnapshot();
});

it("should run callback when clicked", function() {
  const cb = jest.fn();
  const wrapper = shallow(
    <TodoItem {...defaultProps} onClick={cb} />
  );

  var listItemEl = wrapper.find("li");
  listItemEl.simulate("click");

  expect(cb).toHaveBeenCalledTimes(1);
  cb(42);
  cb(52, 11);
  expect(cb.mock.calls).toMatchSnapshot();
});
