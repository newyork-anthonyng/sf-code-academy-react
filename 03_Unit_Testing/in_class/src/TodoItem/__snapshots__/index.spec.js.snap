// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have correct movie 1`] = `
Array [
  Object {
    "ageRating": 12,
    "name": "The Dark Knight",
    "rating": 7,
  },
  Object {
    "ageRating": 14,
    "name": "Up",
    "rating": 10,
  },
]
`;

exports[`should render correctly when completed 1`] = `
<li
  onClick={[Function]}
  style={
    Object {
      "color": "green",
      "cursor": "pointer",
      "textDecoration": "line-through",
    }
  }
>
  Walk dog
</li>
`;

exports[`should render correctly when not completed 1`] = `
<li
  onClick={[Function]}
  style={
    Object {
      "color": "black",
      "cursor": "pointer",
      "textDecoration": "",
    }
  }
>
  Walk dog
</li>
`;

exports[`should render correctly when not completed 2`] = `
Object {
  "foo": "bar",
}
`;

exports[`should run callback when clicked 1`] = `
Array [
  Array [
    42,
  ],
  Array [
    42,
  ],
  Array [
    52,
    11,
  ],
]
`;
